### ISSUE TITLE
Can't deactivate UseIfInsteadOfWhen rule

### ISSUE THREAD

**Issue by sschuberth** [arturbosch/detekt#1986]
## Expected Behavior

I'd expect this to deactivate the `UseIfInsteadOfWhen` rule:

```
style:
  UseIfInsteadOfWhen:
    active: false
```

## Observed Behavior

Deactivating the rule has no effect, such violations are still reported.

## Steps to Reproduce

* Clone https://github.com/heremaps/oss-review-toolkit and checkout the `detekt-issue-1986` branch.
* Run `./gradlew :utils:detekt`.
* Observe:

```
Ruleset: style - 5min debt
        UseIfInsteadOfWhen - [spdxExpression] at C:\Dev\oss-review-toolkit\utils\src\main\kotlin\DeclaredLicenseProcessor.kt:77:30
```

## Context

I guess this happens because `config` is not passed at https://github.com/arturbosch/detekt/blob/d95d7d1b1f0d0f36fc74dd5bfcfd00ac2f113fee/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/providers/StyleGuideProvider.kt#L120.

## Your Environment

* Version of detekt used: 1.1.0
* Version of Gradle used (if applicable): 5.6.2
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): https://github.com/heremaps/oss-review-toolkit

**Comment by schalkms** [comment]
Thanks for reporting this!
